modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -6057,6 +6058,183 @@ public void testSearchObjectEntriesWithAccountEntryRestricted()
+				objectDefinition.getCompanyId(), adminUser.getUserId(),
+				objectDefinition.getClassName(), objectEntry2.getId()));
+		Assert.assertTrue(
+			_subscriptionLocalService.isSubscribed(
+				objectDefinition.getCompanyId(), _user.getUserId(),
+				objectDefinition.getClassName(), objectEntry2.getId()));
+
+		objectDefinitionLocalService.deleteObjectDefinition(objectDefinition);
+
+		Assert.assertFalse(
+			_subscriptionLocalService.isSubscribed(
+				objectDefinition.getCompanyId(), adminUser.getUserId(),
+				objectDefinition.getClassName(), objectEntry2.getId()));
+		Assert.assertFalse(
+			_subscriptionLocalService.isSubscribed(
+				objectDefinition.getCompanyId(), _user.getUserId(),
+				objectDefinition.getClassName(), objectEntry2.getId()));
+	}
+
+	@FeatureFlag("LPD-42577")
+	@Test
+	public void testSubscribeAndUnsubscribeObjectEntryWithHierarchy()
+		throws Exception {
+
+		ObjectDefinition objectDefinitionA =
+			ObjectDefinitionTestUtil.publishObjectDefinition();
+		ObjectDefinition objectDefinitionAA =
+			ObjectDefinitionTestUtil.publishObjectDefinition();
+
+		ObjectRelationship objectRelationship =
+			_objectRelationshipLocalService.addObjectRelationship(
+				null, adminUser.getUserId(),
+				objectDefinitionA.getObjectDefinitionId(),
+				objectDefinitionAA.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_CASCADE, true,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				StringUtil.randomId(), false,
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY, null);
+
+		ObjectEntry rootObjectEntry = _addObjectEntry(
+			objectDefinitionA, Collections.emptyMap());
+
+		_defaultObjectEntryManager.subscribeObjectEntry(
+			_simpleDTOConverterContext,
+			rootObjectEntry.getExternalReferenceCode(), objectDefinitionA,
+			rootObjectEntry.getScopeKey());
+
+		Assert.assertTrue(
+			_subscriptionLocalService.isSubscribed(
+				objectDefinitionA.getCompanyId(), adminUser.getUserId(),
+				objectDefinitionA.getClassName(), rootObjectEntry.getId()));
+
+		_defaultObjectEntryManager.unsubscribeObjectEntry(
+			_simpleDTOConverterContext,
+			rootObjectEntry.getExternalReferenceCode(), objectDefinitionA,
+			rootObjectEntry.getScopeKey());
+
+		Assert.assertFalse(
+			_subscriptionLocalService.isSubscribed(
+				objectDefinitionA.getCompanyId(), adminUser.getUserId(),
+				objectDefinitionA.getClassName(), rootObjectEntry.getId()));
+
+		ObjectField objectField = _objectFieldLocalService.getObjectField(
+			objectRelationship.getObjectFieldId2());
+
+		ObjectEntry childObjectEntry = _addObjectEntry(
+			objectDefinitionAA,
+			Collections.singletonMap(
+				objectField.getName(), rootObjectEntry.getId()));
+
+		AssertUtils.assertFailure(
+			UnsupportedOperationException.class, null,
+			() -> _defaultObjectEntryManager.subscribeObjectEntry(
+				_simpleDTOConverterContext,
+				childObjectEntry.getExternalReferenceCode(), objectDefinitionAA,
+				childObjectEntry.getScopeKey()));
+	}
```

Comment:
@vngoliveira please delete `objectDefinitionA` and `objectDefinitionAA` at the end of this test.